File: tests/e2e/fixtures/java/BreakpointTest.java
/**
 * Breakpoint test program for E2E testing
 * Tests breakpoint functionality
 */
public class BreakpointTest {
    private int counter = 0;

    public static void main(String[] args) throws Exception {
        BreakpointTest test = new BreakpointTest();
        test.run();
    }

    public void run() throws Exception {
        System.out.println("Starting breakpoint test");

        // Line 15: First breakpoint location
        methodA();

        // Line 18: Second breakpoint location
        methodB();

        // Line 21: Third breakpoint location
        methodC();

        System.out.println("Breakpoint test completed");
    }

    private void methodA() {
        counter++; // Line 27
        System.out.println("methodA: counter = " + counter);
    }

    private void methodB() {
        counter += 2; // Line 32
        System.out.println("methodB: counter = " + counter);
    }

    private void methodC() {
        counter += 3; // Line 37
        System.out.println("methodC: counter = " + counter);
    }
}

File: tests/e2e/fixtures/go/breakpoint_test.go
// Breakpoint test program for E2E testing
package main

import (
	"fmt"
	"time"
)

// BreakpointTest is a struct for testing method breakpoints
type BreakpointTest struct {
	counter int
}

// MethodA is a method to set breakpoint on
func (b *BreakpointTest) MethodA() {
	b.counter++ // Breakpoint location 1
	fmt.Printf("MethodA: counter = %d\n", b.counter)
}

// MethodB is another method to set breakpoint on
func (b *BreakpointTest) MethodB() {
	b.counter += 2 // Breakpoint location 2
	fmt.Printf("MethodB: counter = %d\n", b.counter)
}

func main() {
	test := &BreakpointTest{counter: 0}

	fmt.Println("Starting breakpoint test")

	// Short delay for debugger
	time.Sleep(100 * time.Millisecond)

	test.MethodA() // Breakpoint location 3
	test.MethodB() // Breakpoint location 4

	fmt.Println("Breakpoint test completed")

	// Keep running
	time.Sleep(60 * time.Second)
}
